classes/query/QueryDAO.php

```diff
@@ -394,23 +421,71 @@ public function addRecommendationQuery(int $recommenderUserId, int $submissionId
             // Check if the user is unsubscribed
             $notificationSubscriptionSettings = $notificationSubscriptionSettingsDao->getNotificationSubscriptionSettings(
                 NotificationSubscriptionSettingsDAO::BLOCKED_EMAIL_NOTIFICATION_KEY,
-                $discussionParticipantsId,
+                $participantUserId,
                 $contextId
             );
             if (in_array(PKPNotification::NOTIFICATION_TYPE_NEW_QUERY, $notificationSubscriptionSettings)) {
                 continue;
             }
-            $recipient = Repo::user()->get($discussionParticipantsId);
+
+            $recipient = Repo::user()->get($participantUserId);
             $mailable = new Mailable();
             $mailable->to($recipient->getEmail(), $recipient->getFullName());
-            $mailable->from($recommenderUser->getEmail(), $recommenderUser->getFullName());
+            $mailable->from($fromUser->getEmail(), $fromUser->getFullName());
             $mailable->subject($title);
             $mailable->body($content);
+
             Mail::send($mailable);
         }
 
         return $query->getId();
     }
+
+    /**
+     * Create a query with a submission's comments for the editors
+     *
+     * Creates the query and assigns all participants
+     *
+     * @return int new query id
+     */
+    public function addCommentsForEditorsQuery(Submission $submission): int
+    {
+
+        /** @var StageAssignmentDAO $stageAssignmentDao */
+        $stageAssignmentDao = DAORegistry::getDAO('StageAssignmentDAO');
+        $assigned = $stageAssignmentDao->getBySubmissionAndRoleIds(
+            $submission->getId(),
+            [
+                Role::ROLE_ID_MANAGER,
+                Role::ROLE_ID_SUB_EDITOR,
+                Role::ROLE_ID_ASSISTANT,
+                Role::ROLE_ID_AUTHOR,
+            ],
+            $submission->getData('stageId')
+        );
+        $assigned = collect($assigned->toArray());
```

Comment:
Isn't this just round-tripping a collection to an array and back again?

Reply:
In this case, `$assigned` is a `DAOResultFactory` so it's going `DAOResultFactory` -> `array` -> `Collection`.